create_bags/bag_creator.py

```diff
@@ -41,14 +42,16 @@ def construct_job_params(self, rights_ids, files, dates):
             dates (tuple): begin and end dates
 
         Returns a dictionary"""
-        job_params = {"workflowName": "Digitization Workflow"}
-        job_params['packageName'] = "{}.tar".format(self.refid)
-        job_params['files'] = files
+
+        job_params = {
+            "workflowName": "Digitization Workflow",
+            "packageName": "{}.tar".format(
+                self.refid),
+            "files": files}
```

Comment:
```suggestion
        job_params = {
            "workflowName": "Digitization Workflow",
            "packageName": "{}.tar".format(
                self.refid),
            "files": files,
            "tags": [
                {"tagFile": "bag-info.txt", "tagName": "ArchivesSpace-URI", "userValue": self.ao_uri},
                {"tagFile": "bag-info.txt", "tagName": "Start-Date", "userValue": begin_date},
                {"tagFile": "bag-info.txt", "tagName": "End-Date", "userValue": end_date}
            ]}
```
I know this has some repetition in it, but it makes understanding what that data looks like a lot easier.

Reply:
I understand why this is more readable, but I'm really struggling with getting this approach to work with an unknown number of rights IDs. The above combined with
```
for rights_id in rights_ids:
            job_params['tags'].append(create_tag("Rights-ID", str(rights_id)))
```
Is producing:
`[{'tagFile': 'bag-info.txt', 'tagName': 'ArchivesSpace-URI', 'userValue': '/whatever'}, {'tagFile': 'bag-info.txt', 't...txt', 'tagName': 'Rights-ID', 'userValue': '2'}, {'tagFile': 'bag-info.txt', 'tagName': 'Rights-ID', 'userValue': '4'}]`

Reply:
What output are you expecting and what's that `...` in the middle of the dictionary all about?

Reply:
I'm expecting 

```
[{'tagFile': 'bag-info.txt', 'tagName': 'ArchivesSpace-URI', 'userValue': '/whatever'}, {'tagFile': 'bag-info.txt', 'tagName': 'Start-Date', 'userValue': '1940-01-01'}, {'tagFile': 'bag-info.txt', 'tagName': 'End-Date', 'userValue': '1940-06-01'}, {'tagFile': 'bag-info.txt', 'tagName': 'Origin', 'userValue': 'digitization'}, {'tagFile': 'bag-info.txt', 'tagName': 'Rights-ID', 'userValue': '2'}, {'tagFile': 'bag-info.txt', 'tagName': 'Rights-ID', 'userValue': '4'}]
```
And I was wondering the same thing--it's super weird.

Reply:
I am not sure how you're printing that out but I suspect that's an ellipsis...

Maybe a good time to write tests that assert on the output of the params?

Reply:
I found this issue through the tests, which showed that output. I'm not sure how to write a more focused test since that's all happening in that method.

Reply:
I understand this would sacrifice readability, but the original approach did not cause issues so I'm inclined to go with that.

Reply:
You just need to add `{"tagFile": "bag-info.txt", "tagName": "Origin", "userValue": "digitization"}` to the `tags` array - I missed that before.

Pytest is truncating the output - if you want to see full output you should print from a method.

Reply:
A better test would assert on the values of each key in the `tags` list - as it stands now you're just asserting that there are a certain number of keys in that list, but not what those keys or their values are.

Reply:
It's working now--thank you!!